app/views/shared/_default_client_settings.html.erb

```diff
@@ -60,6 +60,13 @@
       settings[:launch_context] = lti.launch_context
       settings[:tool_consumer_instance_guid] = lti.tool_consumer_instance_guid
       settings[:tool_consumer_instance_name] = lti.tool_consumer_instance_name
+
+      if state_validation = request.env["atomic.validated.state_validation"]&.with_indifferent_access
+        settings[:state] = state_validation["state"]
+        settings[:ltiStorageParams] = state_validation["lti_storage_params"]
+        settings[:stateVerified] = state_validation["verified_by_cookie"]
+      end
```

Comment:
How about this:
```
settings[:state_validation] = atomic.validated.state_validation
```

and then later call `ltilaunch(window.DEFAULT_SETTINGS.state_valdation)`

Reply:
I made that change.